### PR TITLE
Add support for nested fields

### PR DESCRIPTION
**Pull request by thani-sh**

### CODE REVIEW

File: lib/pipes/group.js [hunk 1]
```diff
@@ -16,58 +15,59 @@ Pipe.extendWith(Group);
...
-      result.push(groupings[key]);
+  var _idDsl = this.dsl._id;
+
+  docs.forEach(function (doc) {
+    var _id;
+
+    if(typeof _idDsl === 'string') {
+      _id = expr.evalExpr(_idDsl, doc);
+    } else if(typeof _idDsl === 'object') {
+      _id = expr(_idDsl, doc);
```
Comment (arunoda) [inline lib/pipes/group.js]: Add a method like `evalObject`

File: lib/pipes/sort.js [hunk 2]
```diff
@@ -50,4 +54,11 @@ Sort.prototype.apply = function(data) {
   return data;
 };
 
+
+Sort.prototype._sortableValue = function(field, doc) {
+  var value = expr.val('$' + field, doc);
```
Comment (arunoda) [inline lib/pipes/sort.js:59]: check special handling for Date.

**Comment by thani-sh** [comment]
Done

File: lib/pipes/sort.js (hunk 2 above)
Comment (arunoda) [inline lib/pipes/sort.js:59]: Seems like we don't need that.

**Pull request merged**